// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.intel.examples.objectDetection";
option java_outer_classname = "objectDetectionProto";
option objc_class_prefix = "HLW";

package objectDetection;

// The Detection service definition.
service Detection {
  // Send Input Blobs and receive Output Blobs
  rpc getInferResult (RequestDataTensors) returns (ReplyDataTensors) {}
  rpc sendXml (stream RequestDataChunks) returns (ReplyStatus) {}
  rpc sendBin (stream RequestDataChunks) returns (ReplyStatus) {}
  rpc prepare (RequestString) returns (ReplyStatus) {} //Placeholder for any future support : RequestString
}


message RequestDataChunks {
  bytes data = 1;
}

message RequestString {
  string value = 1;
}
message ReplyStatus {
  bool status = 1;
}

// Structure defining the structure of the Data Tensor(blob)
message DataTensor {
  bytes data = 1;
  string node_name = 2;
  repeated int32 tensor_shape = 3;
}

// Reply message containing the Output Data Tensors(blobs)
message ReplyDataTensors {
  repeated DataTensor data_tensors = 1;
}

// Request message containing the Input Data Tensors(blobs)
message RequestDataTensors {
  repeated DataTensor data_tensors = 1;
}